translate/src/core/placeable/components/Highlight.tsx

```diff
@@ -0,0 +1,71 @@
+import { Localized } from '@fluent/react';
+import escapeRegExp from 'lodash.escaperegexp';
+import { cloneElement, ReactNode } from 'react';
+import createMarker, { getRules, Parser } from 'react-content-marker';
+import { TermState } from '~/core/term';
+
+import './Highlight.css';
+
+let keyCounter = 0;
+const MarkerCache = new Map<string, ReturnType<typeof createMarker>>();
+
+/**
+ * Component that marks placeables in a string.
```

Comment:
```suggestion
 * Component that marks placeables and terms in a string.
```

`react-content-marker` doesn't handle wrapping itself really well, so using two separate components would require some further changes there, as well as some way to avoid terms being mistakenly identified within placeables. In other words, it gets Complicated.